content/en/docs/tasks/administer-cluster/migrating-from-dockershim/find-if-you-affected.md

```diff
@@ -0,0 +1,62 @@
+order to understand how your cluster or apps may acquire dependency on Docker as
+a container runtime, you need to understand what role dockershim plays when
+container is being scheduled on the node.
+
+## Container runtimes
+
+[Container runtime](/docs/setup/production-environment/container-runtimes/) is a software that can schedule Pods. Kubernetes is responsible for orchestration of application Pods, and uses CRI (container runtime interface) to communicate to container runtime to actually schedule these Pods and get their status.
+
+There are various container runtimes, each has special features and decide on tradeoffs they make between performance, security and functionality. Dockershim is a first container runtime used by Kubernetes. When dockershim is used as a container runtime, kubelet communicates with Docker to schedule containers and check on it’s status. Docker in its turn uses Containerd to actually schedule containers, while also making these containers visible for Docker tooling and UI.
+
+You can read about it in [kubernetes containerd integration goes GA](/blog/2018/05/24/kubernetes-containerd-integration-goes-ga/) blog post.
+
+![Dockershim vs. CRI with Containerd](/images/blog/2018-05-24-kubernetes-containerd-integration-goes-ga/cri-containerd.png)
+
+Switching to containerd as a container runtime eliminates the middleman. All the
+same containers can be run by container runtimes like Containerd as before. But
+now, since containers were scheduled directly with the container runtime, they
+are not visible to Docker. So any Docker tooling or fancy UI you might have used
+before to check on these containers is no longer available.
+
+You cannot get container information using `docker ps` or `docker inspect`
+commands. As you cannot list containers, you cannot get logs, stop containers,
+or execute something with `docker exec`.
+
+{{< note >}}
+
+Stopping containers via Docker CLI was never a good idea as it is
+better to allow Kubernetes to orchestrate them. So you don’t confuse Kubernetes
+by disappearing containers.
+
+{{< /note >}}
+
+You can still pull images or build them using `docker build` command. But images
+built or pulled by Docker would not be visible to container runtime and
+Kubernetes. They needed to be pushed to some registry to allow them to be used
+by kubernetes.
+
+## Finding if your app has a dependencies on Docker
+
+1. Make sure no privileged Pods executing Docker commands
+2. Check that scripts and apps running on nodes outside of Kubernetes
+   infrastructure do not execute docker commands. It might be:
+   - SSH to nodes to troubleshoot
+   - Node startup scripts
+   - Monitoring and security agents installed on nodes directly
+3. Third-party tools that perform above mentioned privileged operations. See
+   [Migrating telemetry and security agents from
+   dockershim](migrating-telemetry-and-security-agents.md) for more information.
```

Comment:
nit: broken hyperlink

Reply:
do you mean `.md`? Always forgetting. Sorry if it is being annoying.